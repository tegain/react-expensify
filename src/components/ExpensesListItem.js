import React from 'react'
import moment from 'moment'
import numeral from 'numeral'
import '../locales/fr'
import { Link } from 'react-router-dom'

const ExpensesListItem = ({ id, description, amount, createdAt, note }) => (
	<div className="expense-item">
		<Link to={`/edit/${id}`}>
			<div className="expense-item__content">
				<h3 className="expense-item__title">{description}</h3>
				<span className="expense-item__date">{moment(createdAt).format('Do MMMM YYYY')}</span>
				{
					(note) && (
						<p className="expense-item__note">
							{note}
						</p>
					)
				}
			</div>
			<span className="expense-item__amount">
				{numeral(amount / 100).format("0,0[.][00] $")}
			</span>
		</Link>
	</div>
)

export default ExpensesListItem