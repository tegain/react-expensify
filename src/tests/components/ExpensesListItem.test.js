import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import ExpensesListItem from '../../components/ExpensesListItem'

test('should render <ExpensesListItem/> correctly', () => {
	const wrapper = shallow(<ExpensesListItem {...expenses[0]} />)
	expect(wrapper).toMatchSnapshot()
})