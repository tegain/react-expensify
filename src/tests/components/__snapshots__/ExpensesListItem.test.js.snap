// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <ExpensesListItem/> correctly 1`] = `
<div
  className="expense-item"
>
  <Link
    replace={false}
    to="/edit/1"
  >
    <div
      className="expense-item__content"
    >
      <h3
        className="expense-item__title"
      >
        Rent
      </h3>
      <span
        className="expense-item__date"
      >
        1st January 1970
      </span>
    </div>
    <span
      className="expense-item__amount"
    >
      1,95 â‚¬
    </span>
  </Link>
</div>
`;
